include/ringmesh/basic/types.h

```diff
@@ -76,6 +78,39 @@ namespace RINGMesh
     // This is an array of 3 doubles
     using vec2 = vecn< 2 >;
 
+    template < index_t DIMENSION >
+    bool operator==( const vecn< DIMENSION >& u, const vecn< DIMENSION >& v )
+    {
+        for( index_t i = 0; i < DIMENSION; i++ )
+        {
+            if( u[i] != v[i] )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    template < index_t DIMENSION >
+    bool operator!=( const vecn< DIMENSION >& u, const vecn< DIMENSION >& v )
+    {
+        return !( u == v );
+    }
+
+    template < index_t DIMENSION >
+    bool inexact_equal( const vecn< DIMENSION >& v1,
+        const vecn< DIMENSION >& v2,
+        double epsilon )
+    {
+        double length{ 0. };
+        for( index_t i = 0; i < DIMENSION; i++ )
+        {
+            length += ( v2[i] - v1[i] ) * ( v2[i] - v1[i] );
+        }
+        length = std::sqrt( length );
```

Comment:
instead of computing sqrt, you can compare the square_length with epsilon * epsilon

Reply:
True, that makes me think it will also remove the test of epsilon > 0.